course_catalog/task_helpers_test.py

```diff
@@ -455,3 +264,51 @@ def test_safe_load_bad_json(mocker):
     mock_logger = mocker.patch("course_catalog.task_helpers.log.exception")
     assert safe_load_json("badjson", "key") == {}
     mock_logger.assert_called_with("%s has a corrupted JSON", "key")
+
+
+@pytest.mark.parametrize(
+    "course_id,course_json,platform, expected",
+    [
+        [
+            "MITX-01",
+            {"course_runs": [{"marketing_url": "https://www.edx.org/course/someurl"}]},
+            PlatformType.mitx.value,
+            "https://www.edx.org/course/someurl",
+        ],
+        [
+            "MITX-01",
+            {"course_runs": [{"marketing_url": "https://www.edx.org/"}]},
+            PlatformType.mitx.value,
+            "https://courses.edx.org/courses/MITX-01/course/",
+        ],
+        [
+            "MITX-01",
+            {"course_runs": [{"marketing_url": ""}]},
+            PlatformType.mitx.value,
+            "https://courses.edx.org/courses/MITX-01/course/",
+        ],
+        [
+            "MITX-01",
+            {"course_runs": [{}]},
+            PlatformType.mitx.value,
+            "https://courses.edx.org/courses/MITX-01/course/",
+        ],
+        [
+            "MITX-01",
+            {},
+            PlatformType.mitx.value,
+            "https://courses.edx.org/courses/MITX-01/course/",
+        ],
+        [
+            "e9387c256bae4ca99cce88fd8b7f8272",
+            {"url": "/someurl"},
+            PlatformType.ocw.value,
+            "http://ocw.mit.edu/someurl",
+        ],
+        ["e9387c256bae4ca99cce88fd8b7f8272", {"url": ""}, PlatformType.ocw.value, None],
+        ["e9387c256bae4ca99cce88fd8b7f8272", {}, PlatformType.ocw.value, None],
+    ],
+)
+def test_get_course_url(course_id, course_json, platform, expected):
+    """ Test that url's are calculated as expected """
+    assert get_course_url(course_id, course_json, platform) == expected
```

Comment:
This will compare `None == None` on the last two tests and it should be `is expected` for those ones.